haskell/repl.bzl

```diff
@@ -174,114 +173,14 @@ def _merge_HaskellReplCollectInfo(args):
         transitive_cc_dependencies = transitive_cc_dependencies,
     )
 
-def _parse_pattern(pattern_str):
```

Comment:
Ignore most of my previous comments, this PR is not supposed to clean `parse_pattern` at the same time it moves it.

Reply:
If you want to do that, you can do so in a different commit. Might be best to split the move from the other stuff anyway.

Reply:
@guibou I resolved your comments re: pattern parsing, since they're out of scope for this PR. I think there will be some work on that code in another PR, so I'll keep them in mind.